projects/common/ctpnxe/ctpnxe_system_bd_mc.tcl

```diff
@@ -0,0 +1,316 @@
+  "$project_name/cpu0_inst/IRQ_S4"
+sbp_connect_interface_net "$project_name/uart0_inst/INT_M0" \
+  "$project_name/cpu0_inst/IRQ_S0"
+sbp_connect_interface_net "$project_name/spi0_inst/INTR" \
+  "$project_name/cpu0_inst/IRQ_S2"
+sbp_connect_interface_net "$project_name/i2c0_inst/INTR" \
+  "$project_name/cpu0_inst/IRQ_S3"
+
+sbp_assign_addr_seg -offset 'h00000000 "$project_name/ahbl0_inst/AHBL_M00" \
+  "$project_name/isr_ram0_inst/AHBL_S0"
+sbp_assign_addr_seg -offset 'h000C0000 "$project_name/ahbl0_inst/AHBL_M01" \
+  "$project_name/sysmem0_inst/AHBL_S0"
+sbp_assign_addr_seg -offset 'h000E0000 "$project_name/ahbl0_inst/AHBL_M04" \
+  "$project_name/sysmem0_inst/AHBL_S1"
+sbp_assign_addr_seg -offset 'h00186000 "$project_name/apb0_inst/APB_M02" \
+  "$project_name/i2c0_inst/APB_S0"
+sbp_assign_addr_seg -offset 'h00186400 "$project_name/apb0_inst/APB_M00" \
+  "$project_name/uart0_inst/APB_S0"
+sbp_assign_addr_seg -offset 'h00186800 "$project_name/ahbl0_inst/AHBL_M03" \
+  "$project_name/spi0_inst/AHBL_S0"
+sbp_assign_addr_seg -offset 'h00186C00 "$project_name/apb0_inst/APB_M01" \
+  "$project_name/gpio0_inst/APB_S0"
+sbp_assign_addr_seg -offset 'h00187000 "$project_name/apb0_inst/APB_M03" \
+  "$project_name/gpio1_inst/APB_S0"
+
+## You can use this part in your project to instantiate a timer ip with max
+# 8 timers if you need it.
+
+# adi_ip_update -vlnv {latticesemi.com:ip:cpu0:2.5.0} \
+#   -meta_vlnv {latticesemi.com:ip:riscv_mc:2.5.0} \
+#   -cfg_value {
+#     SIMULATION: false,
+#     DEBUG_ENABLE: true,
+#     M_STANDALONE: true,
+#     IRQ_NUM: 6
+#   } \
+#   -ip_iname "cpu0_inst"
+
+# adi_ip_update -vlnv {latticesemi.com:module:apb0:1.2.0} \
+#   -meta_vlnv {latticesemi.com:module:apb_interconnect:1.2.0} \
+#   -cfg_value {
+#     TOTAL_MASTER_CNT: 1,
+#     TOTAL_SLAVE_CNT: 5
+#   } \
+#   -ip_iname "apb0_inst"
+
+# adi_ip_instance -vlnv {latticesemi.com:ip:timer0:1.3.0} \
+#   -meta_vlnv {latticesemi.com:ip:gp_timer:1.3.0} \
+#   -cfg_value {
+#     t1_cnt_up: count-up,
+#     T1_PERIOD_WIDTH: 32,
+#     t1_period_val: 0,
+#     t1_dis_pscaler: false,
+#     t2_cnt_up: count-up,
+#     T2_PERIOD_WIDTH: 32,
+#     t2_period_val: 0,
+#     t2_dis_pscaler: false,
+#     t3_cnt_up: count-up,
+#     T3_PERIOD_WIDTH: 32,
+#     t3_period_val: 0,
+#     t4_cnt_up: count-up,
+#     T4_PERIOD_WIDTH: 32,
+#     t4_period_val: 0
+#   } \
+#   -ip_iname "timer0_inst"
+
+# sbp_connect_net -name [sbp_get_nets -from $project_name/pll0_inst *clkop*] \
+#   "$project_name/timer0_inst/clk_i"
+
+# sbp_connect_net -name [sbp_get_nets -from $project_name/cpu0_inst *resetn*] \
+#   "$project_name/timer0_inst/rst_n_i"
+
+# sbp_connect_interface_net "$project_name/apb0_inst/APB_M04" \
+#   "$project_name/timer0_inst/APB_S0"
+
+# sbp_connect_interface_net "$project_name/timer0_inst/INTR" \
+#   "$project_name/cpu0_inst/IRQ_S5"
+
+# sbp_assign_addr_seg -offset 'h00187400 "$project_name/apb0_inst/APB_M04" \
+#   "$project_name/timer0_inst/APB_S0"
```

Comment:
We don't want commented code in the sources. Do we want to add the timer to the base design or not ?

Reply:
Maybe we could remove this whole design if we only use the riscv-rx CPU.

Reply:
Solved in this  [commit](https://github.com/analogdevicesinc/hdl/pull/1270/commits/05015858329de4d1b632e5e6cdc0e36c9f6550c4).